app/routes/ap-listing.js

```diff
@@ -0,0 +1,95 @@
+const { holdAdmin, schemeAdmin, dataView } = require('../auth/permissions')
+const api = require('../api')
+const convertToCSV = require('../convert-to-csv')
+const apListingSchema = require('../routes/schemas/ap-listing-schema')
+
+module.exports = [
+  {
+    method: 'GET',
+    path: '/report-list/ap-listing',
+    options: {
+      auth: { scope: [holdAdmin, schemeAdmin, dataView] },
+      handler: async (request, h) => {
+        return h.view('ap-listing-report')
+      }
+    }
+  },
+  {
+    method: 'GET',
+    path: '/report-list/ap-listing/download',
+    options: {
+      auth: { scope: [holdAdmin, schemeAdmin, dataView] },
+      validate: {
+      query: apListingSchema,
+        failAction: async (request, h, err) => {
+          request.log(['error', 'validation'], err)
+          const errors = err.details ? err.details.map(detail => {
+            return {
+              text: detail.message,
+              href: '#' + detail.path[0]
+            }
+          }) : []
+          const data = { errors }
+          return h.view('ap-listing-report', data).code(400).takeover()
+        }
+    },
+      handler: async (request, h) => {
+        const { 'start-date-day': startDay, 'start-date-month': startMonth, 'start-date-year': startYear, 'end-date-day': endDay, 'end-date-month': endMonth, 'end-date-year': endYear } = request.query
+
+        let url = '/ap-report-data'
+        let startDate, endDate
+
+        if (startDay && startMonth && startYear) {
+          startDate = `${startYear}-${String(startMonth).padStart(2, '0')}-${String(startDay).padStart(2, '0')}`
+        }
+
+        if (endDay && endMonth && endYear) {
+          endDate = `${endYear}-${String(endMonth).padStart(2, '0')}-${String(endDay).padStart(2, '0')}`
+        } else if (startDate) {
+          const now = new Date()
+          endDate = `${now.getFullYear()}-${String(now.getMonth() + 1).padStart(2, '0')}-${String(now.getDate()).padStart(2, '0')}`
+        }
```

Comment:
do you need an equivalent to this to allow them to provide an end date but not a start date? or do we just not permit that behaviour? 

could in theory just set start date to 01/01/2015 to make sure we gather all the data we have. but depends if we want to allow them to do that?

Reply:
The behaviour is exactly like this! There are 4 ways of downloading the report :
1. Full report, when user provides no dates in submit form.
2. From 01.01.2015 until user input
3. From user input until today
4. From user input until user input

Let me know if you want me to alter this behaviour, however I feel this is a reasonable functionality

Reply:
this is handled on tracking-side, if no values are provided for start date when calling API, the value will be set as null, which will then pull all records from database until date provided (based on our conversation that there wont be any records prior to 2015). Do you want me to instead hard code 01.01.2015?

Reply:
Okay understand. I think ideally lets do it here instead (or possibly do it here and leave tracking side as is, just in case). Reason being is I think I'd rather our file name had the exact dates rather than e.g. 'beginning', just brings some vagueness we'll inevitably have to answer to someone on.

Reply:
added default value for startDate to be 2015-01-01